src/meta/src/manager/catalog/mod.rs

```diff
@@ -1743,6 +1744,95 @@ impl CatalogManager {
         Ok(version)
     }
 
+    pub async fn alter_owner(
+        &self,
+        entity: Entity,
+        owner_id: UserId,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        let user_core = &mut core.user;
+
+        let notify_info;
+        match entity {
+            Entity::TableId(table_id) => {
+                database_core.ensure_table_id(table_id)?;
+                let mut tables = BTreeMapTransaction::new(&mut database_core.tables);
+                let mut table = tables.get_mut(table_id).unwrap();
+                user_core.decrease_ref(table.owner);
```

Comment:
Need to call `decrease_ref` after commit meta is complete to ensure consistency.